hid-xpadneo/src/hid-xpadneo.c

```diff
@@ -321,31 +381,79 @@ static int xpadneo_ff_play(struct input_dev *dev, void *data, struct ff_effect *
 
 static void xpadneo_welcome_rumble(struct hid_device *hdev)
 {
+	struct xpadneo_devdata *xdata = hid_get_drvdata(hdev);
 	struct ff_report ff_pck;
 
 	memset(&ff_pck.ff, 0, sizeof(ff_pck.ff));
 
 	ff_pck.report_id = XPADNEO_XB1S_FF_REPORT;
-	ff_pck.ff.magnitude_weak = 40;
-	ff_pck.ff.magnitude_strong = 20;
-	ff_pck.ff.magnitude_right = 10;
-	ff_pck.ff.magnitude_left = 10;
-	ff_pck.ff.pulse_sustain_10ms = 5;
-	ff_pck.ff.pulse_release_10ms = 5;
-	ff_pck.ff.loop_count = 3;
-
-	ff_pck.ff.enable = FF_RUMBLE_WEAK;
-	hid_hw_output_report(hdev, (u8 *)&ff_pck, sizeof(ff_pck));
-	mdelay(330);
+	if ((xdata->quirks & XPADNEO_QUIRK_NO_MOTOR_MASK) == 0) {
+		ff_pck.ff.magnitude_weak = 40;
+		ff_pck.ff.magnitude_strong = 20;
+		ff_pck.ff.magnitude_right = 10;
+		ff_pck.ff.magnitude_left = 10;
+	}
+	if ((xdata->quirks & XPADNEO_QUIRK_NO_PULSE) == 0) {
+		ff_pck.ff.pulse_sustain_10ms = 5;
+		ff_pck.ff.pulse_release_10ms = 5;
+		ff_pck.ff.loop_count = 3;
+	}
 
-	ff_pck.ff.enable = FF_RUMBLE_STRONG;
+	if (xdata->quirks & XPADNEO_QUIRK_NO_MOTOR_MASK)
+		ff_pck.ff.magnitude_weak = 40;
+	else
+		ff_pck.ff.enable = FF_RUMBLE_WEAK;
 	hid_hw_output_report(hdev, (u8 *)&ff_pck, sizeof(ff_pck));
-	mdelay(330);
+	mdelay(300);
+	if (xdata->quirks & XPADNEO_QUIRK_NO_MOTOR_MASK)
+		ff_pck.ff.magnitude_weak = 0;
+	else
+		ff_pck.ff.enable = 0;
+	if (xdata->quirks & XPADNEO_QUIRK_NO_PULSE) {
+		u8 save = ff_pck.ff.magnitude_weak;
+		hid_hw_output_report(hdev, (u8 *)&ff_pck, sizeof(ff_pck));
+		ff_pck.ff.magnitude_weak = save;
+	}
+	mdelay(30);
 
-	if (likely(param_trigger_rumble_mode < PARAM_TRIGGER_RUMBLE_DISABLE)) {
-		ff_pck.ff.enable = FF_RUMBLE_TRIGGERS;
+	if (xdata->quirks & XPADNEO_QUIRK_NO_MOTOR_MASK)
+		ff_pck.ff.magnitude_strong = 20;
+	else
+		ff_pck.ff.enable = FF_RUMBLE_STRONG;
+	hid_hw_output_report(hdev, (u8 *)&ff_pck, sizeof(ff_pck));
+	mdelay(300);
+	if (xdata->quirks & XPADNEO_QUIRK_NO_MOTOR_MASK)
+		ff_pck.ff.magnitude_strong = 0;
+	else
+		ff_pck.ff.enable = 0;
+	if (xdata->quirks & XPADNEO_QUIRK_NO_PULSE) {
+		u8 save = ff_pck.ff.magnitude_strong;
 		hid_hw_output_report(hdev, (u8 *)&ff_pck, sizeof(ff_pck));
-		mdelay(330);
+		ff_pck.ff.magnitude_strong = save;
+	}
+	mdelay(30);
+
+	if ((xdata->quirks & XPADNEO_QUIRK_NO_TRIGGER_RUMBLE) == 0) {
+		if (xdata->quirks & XPADNEO_QUIRK_NO_MOTOR_MASK) {
+			ff_pck.ff.magnitude_left = 10;
+			ff_pck.ff.magnitude_right = 10;
+		} else
+			ff_pck.ff.enable = FF_RUMBLE_TRIGGERS;
```

Comment:
The style guide says if either an `if` or an `else` have braces, then both have to have them.